sympy/series/limits.py

```diff
@@ -88,10 +88,15 @@ class Limit(Expr):
     """
 
     def __new__(cls, e, z, z0, dir="+"):
+        temp = e
         e = sympify(e)
+        try:
+            if e == temp:
+                e = ratsimp(e)
+        except Exception :
```

Comment:
Please don't add a generic exception handler. What is the specific exception you are trying to catch here?